synapse/types.py

```diff
@@ -380,29 +381,24 @@ def map_username_to_mxid_localpart(
             onto different mxids
 
     Returns:
-        unicode: string suitable for a mxid localpart
+        string suitable for a mxid localpart
     """
     if not isinstance(username, bytes):
         username = username.encode("utf-8")
 
     # first we sort out upper-case characters
     if case_sensitive:
 
-        def f1(m):
+        def f1(m: Match) -> bytes:
             return b"_" + m.group().lower()
 
         username = UPPER_CASE_PATTERN.sub(f1, username)
     else:
         username = username.lower()
 
-    # then we sort out non-ascii characters
-    def f2(m):
-        g = m.group()[0]
```

Comment:
`m.group()` would be something like `b"x"` (which is equivalent to `"x"` on Python 2).

On Python 2, `g` would be something like `"x"`, so you call `ord(g)` to get `120`.

On Python 3, `g` would be something like `120` directly.